integration-tests/src/test/java/com/datastax/oss/driver/api/core/metadata/SerializableMetadataIT.java

```diff
@@ -0,0 +1,153 @@
+import org.junit.experimental.categories.Category;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+
+@Category(ParallelizableTests.class)
+public class SerializableMetadataIT {
+
+  private static final CcmRule ccmRule = CcmRule.getInstance();
+
+  private static final SessionRule<CqlSession> sessionRule = SessionRule.builder(ccmRule).build();
+
+  @ClassRule public static final TestRule chain = RuleChain.outerRule(ccmRule).around(sessionRule);
+
+  @Before
+  public void setupKeyspace() {
+    List<String> statements =
+        ImmutableList.of(
+            "CREATE TABLE IF NOT EXISTS all (val int, doub int, str text, primary key (val,doub)) WITH CLUSTERING ORDER BY (doub DESC)",
+            "CREATE TYPE IF NOT EXISTS numbers_data (doub int, str text)",
+            "CREATE TABLE IF NOT EXISTS smaller (val int primary key, data numbers_data)",
+            "CREATE INDEX IF NOT EXISTS doub_idx on all (doub)",
+            "CREATE OR REPLACE FUNCTION avgState ( state tuple<int,bigint>, val int ) CALLED ON NULL INPUT RETURNS tuple<int,bigint> LANGUAGE java AS"
+                + " 'if (val !=null) { state.setInt(0, state.getInt(0)+1); state.setLong(1, state.getLong(1)+val.intValue()); } return state;'",
+            "CREATE OR REPLACE FUNCTION avgFinal ( state tuple<int,bigint> ) CALLED ON NULL INPUT RETURNS double LANGUAGE java AS"
+                + " 'double r = 0; if (state.getInt(0) == 0) return null; r = state.getLong(1); r/= state.getInt(0); return Double.valueOf(r);'",
+            "CREATE AGGREGATE IF NOT EXISTS average ( int ) SFUNC avgState STYPE tuple<int,bigint> FINALFUNC avgFinal INITCOND (0,0)",
+            "TRUNCATE all",
+            "INSERT INTO all (val,doub,str) VALUES (1,2,'one')",
+            "INSERT INTO all (val,doub,str) VALUES (2,4,'two')",
+            "INSERT INTO all (val,doub,str) VALUES (3,6,'three')",
+            "CREATE MATERIALIZED VIEW IF NOT EXISTS big AS SELECT * FROM all"
+                + " WHERE val is not null and doub is not null and val > 2"
+                + " PRIMARY KEY (val,doub)");
+
+    CqlSession session = sessionRule.session();
+    statements.forEach(session::execute);
+  }
+
+  private KeyspaceMetadata getAndValidateKsMetadata() {
+
+    Optional<KeyspaceMetadata> ksOption =
+        sessionRule.session().getMetadata().getKeyspace(sessionRule.session().getKeyspace().get());
+    assertTrue(ksOption.isPresent());
+    KeyspaceMetadata ks = ksOption.get();
+    assertTrue(ks instanceof DefaultKeyspaceMetadata);
+    return ks;
+  }
+
+  @Test
+  public void should_contain_serializable_metadata() {
+
+    KeyspaceMetadata ks = getAndValidateKsMetadata();
+
+    /* Validate that the keyspace metadata is fully populated */
+    assertFalse(ks.getUserDefinedTypes().isEmpty());
+    assertFalse(ks.getTables().isEmpty());
+    assertFalse(ks.getViews().isEmpty());
+    assertFalse(ks.getFunctions().isEmpty());
+    assertFalse(ks.getAggregates().isEmpty());
```

Comment:
Nit: we've pretty much settled on AssertJ as de facto standard by now, please use it instead of JUnit assertions. As a bonus, it has more fluent alternatives for most of those checks:
```java
assertThat(ks.getUserDefinedTypes()).isEmpty();
```
+ the optional example from my other comment

Reply:
Bah, that's on me.  I know AssertJ is preferred... I used the JUnit checks when I was getting this test going and forgot to convert before I threw this open to review. :facepalm: 

Will fix.